notify/email/email.go

```diff
@@ -16,15 +16,20 @@ package email
 import (
 	"bytes"
 	"context"
+	"crypto/rand"
```

Comment:
I think that `math/rand` is enough for what we need here. I would add a `generator *rand.Rand` field to the `Email` struct as well as a `hostname string` field (so `os.Hostname()` is called only once). Then `generateMessageID()` would become a method of `*Email`.

Reply:
- `generateMessageID()` is not needed anymore
- add `os.Hostname()` to `func New`